### PR TITLE
Generic-to-IL: C/C++: Declarations inside if conditions

### PR DESCRIPTION
**Pull request by IagoAbal**
Translate if-conditions such as `if (int x = f())` to the Dataflow IL.

Closes PA-3107
Closes PA-3112

test plan:
make test # new test

### CODE REVIEW

File: tests/rules/taint_decl_in_if_cond1.cpp
Comment (amchiclet) [inline tests/rules/taint_decl_in_if_cond1.cpp:1]: Optional: maybe rename file to taint_decl_in_loop_cond.cpp or taint_decl_in_while_cond.cpp, etc.

File: tests/rules/taint_decl_in_if_cond1.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    // sanitize `data`
+    maybe_realloc_data(&data);
+    // ok: source-sink-while
+    sink(data);
+}
+
+void test2() {
+    void *data = get_data();
+
+    // BUG: Declaration in `while` loop does not sanitize
```
Comment (amchiclet) [inline tests/rules/taint_decl_in_if_cond1.cpp]: This comment makes me think that it's a TODO that will be fixed later. Is the comment here to indicate that this used to be a bug but it's fixed now?

**Review by amchiclet — APPROVED** [review]

**Review by ihji — APPROVED** [review]
LGTM

File: tests/rules/taint_decl_in_if_cond1.cpp (hunk 1 above)
Comment (IagoAbal) [reply tests/rules/taint_decl_in_if_cond1.cpp]: I just copy-pasted the example as it was reported, but I get your point, I'm fixing it.

**Pull request merged**